packages/block-editor/src/components/contrast-checker/index.js

```diff
@@ -6,21 +6,21 @@ import tinycolor from 'tinycolor2';
 /**
  * WordPress dependencies
  */
-import { speak } from '@wordpress/a11y';
 import { __ } from '@wordpress/i18n';
 import { Notice } from '@wordpress/components';
-import { useEffect } from '@wordpress/element';
 
-function ContrastCheckerMessage( { tinyBackgroundColor, tinyTextColor, backgroundColor, textColor } ) {
+function ContrastCheckerMessage( { tinyBackgroundColor, tinyTextColor } ) {
 	const msg = tinyBackgroundColor.getBrightness() < tinyTextColor.getBrightness() ?
 		__( 'This color combination may be hard for people to read. Try using a darker background color and/or a brighter text color.' ) :
 		__( 'This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color.' );
-	useEffect( () => {
-		speak( __( 'This color combination may be hard for people to read.' ) );
-	}, [ backgroundColor, textColor ] );
```

Comment:
One thing that I am seeing in rebasing this: Previously, we would speak this message whenever the background or text color would change, which I assume would be useful as an indication of "the new selection is _still_ hard to read". With the changes proposed here, the message would only be spoken when the notice is first shown.

I'm not sure it's a common use-case that `Notice` needs to support, but at least for how we're using it here, we might want to either:

- Revert the changes to this specific component (it was intended as a refactor anyways) and consider it fine if a component wants more granular control around when `speak` is called.
- Or: Change the implementation such that we force `Notice` to remount, or "flip" the prop value of `spokenMessage` ('This color combination ... ' -> undefined -> 'This color combination ... ').

The first of these is simplest, at least in an effort to avoid blocking the rest of the pull request.

Reply:
For posterity's sake, this is the commit I am going to drop from the rebase, in case we want to cherry-pick or refer to it in the future: 4b237afcb3fef299b4f3d22ba8e47b16378230d3

Reply:
I also added an inline code comment in the rebased 86a1790dc2c52fcb5aaf4afab8246a2eec6b1e96 to clarify this requirement, in case it catches the attention of some future maintainer.

Reply:
Added test case in 8ce0daeeb. And glad I did, because while technically this is a refactoring, since the new default behavior of `Notice` would speak its rendered children, the `spokenMessage` must be unset when trying to override it in favor of a custom implementation. This is fixed and tested in 737bfef6d.